client/driver/docker.go

```diff
@@ -1047,11 +1047,12 @@ func (d *DockerDriver) createContainerConfig(ctx *ExecContext, task *structs.Tas
 	}
 
 	config := &docker.Config{
-		Image:     d.imageID,
-		Hostname:  driverConfig.Hostname,
-		User:      task.User,
-		Tty:       driverConfig.TTY,
-		OpenStdin: driverConfig.Interactive,
+		Image:       d.imageID,
+		Hostname:    driverConfig.Hostname,
+		User:        task.User,
+		Tty:         driverConfig.TTY,
+		OpenStdin:   driverConfig.Interactive,
+		StopTimeout: int(task.KillTimeout),
```

Comment:
Needs to be [`task.KillTimeout.Seconds()`](https://golang.org/pkg/time/#Duration.Seconds) as golang Durations are in nanoseconds.

Reply:
Sounds good- I wasn't sure whether casting to an int would be safe.

Reply:
Unfortunately `task.KillTimeout.Seconds()` is of type float64 while StopTimeout is of type int: https://github.com/fsouza/go-dockerclient/pull/693/files#diff-4042076be99ff99c0353b4d447624156R303

Reply:
Wrapping it in `int(...)` should convert it just fine (the loss of precision will be meaningless in this case).

Reply:
👍 thanks for the clarification!

Reply:
See here: https://github.com/hashicorp/nomad/pull/3601/commits/584644287e0624eec4a06ab04606a648cb2af93f